ls1intum/Artemis
# `Programming exercises`: Improve http header for csrf protection for Bamboo

### Checklist
#### General


- [X] Tested on a test server
#### Server

### Motivation and Context


Currently, Artemis has an issue with the CSFR protection of Bamboo. When the student tries to start the exercise, Bamboo returns an error that a Cookie was not set.

### Description

I implemented this workaround: https://confluence.atlassian.com/bamkb/rest-api-calls-fail-due-to-missing-xsrf-token-899447048.html

After testing the issue was resolved and the programming workflow worked again.

### Steps for Testing

As it's a bit difficult to schedule activating this on the actual instance, please test this locally:
Prerequisites:
- 1 Instructor
- 1 Student

1. Checkout the `develop` branch
2. Go to Bamboo > Settings (Cogwheel) > Overview > Security > Security Settings
3. Click below on edit and check `Enable XSFR protection`. Save.
4. Create a programming exercise (instructor)
5. Try to start the exercise (student). It should fail.
6. Checkout this branch (don't forget to restart the server)
7. Create a new programming exercise (instructor)
8. Try to start the exercise (student). Now it should work.
9. Try to participate in the exercise
10. Go to participations as an instructor and resubmit the submission.
11. The workflow should work without issues. Feel free to try further things that affect Bamboo in some way.

### Review Progress





#### Code Review
#### Manual Tests
